backend/src/contexts/users/actions/SignInAction.ts

```diff
@@ -18,4 +18,13 @@ export class SignInAction {
     if (match) return user;
     else throw new WrongPasswordException();
   }
+
+  async executeWithEmail(email: string, password: string) {
```

Comment:
I think maybe it's interesting to put this behavoiur in the execute function, to don't duplicate a lot lines here and in the app file.